zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -371,6 +367,7 @@ impl TransactionRecordsById {
             .iter()
             .for_each(|t| println!("Removing expired mempool tx {}", t));
 
+        // doesnt actually remove the TransactionRecord. all this does is change the spend status to not spent. this is broken because SpendStatus contains a duplicate source of truth as to the ConfirmationStatus of the spending transaction.
```

Comment:
this comment is wrong. it does also delete the record. we will do work in the future to decouple spend status but for now its stable